js/private/utils.bzl

```diff
@@ -0,0 +1,115 @@
+"Utility functions for js rules"
```

Comment:
utils.bzl is a slipperly slope to containing anything. How about bash.bzl?

Reply:
or, does it make sense for it to live near the BASH_RLOCATION_FUNCTION function in bazel_lib?

Reply:
bash.bzl sgtm; this snippet is not fully generalized yet as it depends on a logf_fatal function being available. could be hoisted to bazel-lib in the future once generalized all the way